### PR TITLE
Support stdin-/out forwarding

### PR DESCRIPTION
**Pull request by gillima**
By adding support for stdin-/out forwarding to the Cumulosity server it can be used directly as a ProxyCommand without the need of starting a local TCP server and using `socat`. Also this method works for both, windows and unix.
As `ProxyCommand` is not supporting user interaction, this method requires a active Cumulosity session and will fail otherwise without retry.

### CODE REVIEW

**Comment by switschel** [comment]
Hi @gillima ,

thank you very much for your contribution! This is a helpful feature when you want to use ProxyCommand of ssh.
It seems to be me pretty hard to test but do you see any chance to add an integration test to test the basic behavior of stdio using c8ylp server? https://github.com/SoftwareAG/cumulocity-remote-access-local-proxy/tree/main/tests_integration

**Comment by gillima** [comment]
I've created a "copy" of the `test_ssh_command_then_exit` integration test which now uses the `ssh` executable with `ProxyCommand`. Let me know if you think it has to be improved but as it tests the forwarding I think it does the job...

**Comment by switschel** [comment]
Hi @gillima 
from my site everything looks fine now and it is mergeable, but I leave it to @reubenmiller to have the final decision.

**Comment by reubenmiller** [comment]
Hey sorry for the delay, I will try to run the integrations tests against a real device either today or tomorrow (as the integration tests are not run in gha due to lack of a real test device)

File: tests_integration/test_ssh.py
Comment (reubenmiller) [inline tests_integration/test_ssh.py:55]: Could you please add the following ssh option to the command, otherwise ssh will prompt for the fingerprint when running automated tests.

```python
"-o",
"StrictHostKeyChecking=no",
```

**Comment by reubenmiller** [comment]
Once the small change is done then we can merge this PR (see comment above). All integration tests passed :)

**Review by reubenmiller — APPROVED** [review]

**Comment by reubenmiller** [comment]
@gillima Thanks for the great PR. I did not actually notice the real power of this PR until last night when I was merging it.

I was playing around with the ssh configuration and I realised that the Host can also be a pattern, so I only need to create a common configuration which will be valid for all devices matching that pattern (e.g. perfect for devices which have a common prefix). 

Together with [go-c8y-cli](https://goc8ycli.netlify.app/docs/gettingstarted/) for session management, my workflow now looks like this:

### Setup ssh configuration once (per device type, not per device)

1. Add the following ssh configuration snippet to `./ssh/config` (note: I don't use the --env-file option as I use go-c8y-cli for session/tenant management).

    ```sh
    Host commonprefix-*
        User admin
        PreferredAuthentications publickey
        IdentityFile ~/.ssh/mykey
        ServerAliveInterval 120
        StrictHostKeyChecking no
        UserKnownHostsFile /dev/null
        ProxyCommand python3 -m c8ylp server %n --stdio
    ```

### Connect to devices via ssh

1. Open terminal

2. Switch to a session/tenant using `go-c8y-cli` (which sets all the authorization environment variables that `c8ylp` needs and does TFA prompting where necessary)

    ```sh
    set-session mytenant
    ```

    This is the equivalent of executing `c8ylp login --env-file .env` but it does not a bit nicer of a job switching between different sessions.

4. Connect to device 1 (and do some work)

    ```sh
    ssh commonprefix-device01
    ```

5. Connect to another device (and do some other work)

    ```sh
    ssh commonprefix-device02
    ```

**Disclaimer**

I am the author of [go-c8y-cli](https://goc8ycli.netlify.app/docs/gettingstarted/) which is an opensource cli tool for Cumulocity.

**Comment by gillima** [comment]
@reubenmiller, thanks for the feedback and merging the PR. I will for sure have a look at [go-c8y-cli](https://goc8ycli.netlify.app/docs/gettingstarted/)!

**Pull request merged**
